### DISCUSSION TITLE
No sleep phases?!

### DISCUSSION THREAD

**Discussion by ThisMakesSenseToMe** [Lybron/health-auto-export#24] [Ideas]
For me the most important data to export is Sleep Phases. Will it be supported soon?

**Comment by jwickett** [comment]
sleep phases indeed are supported:

` {
  "metrics": [
    {
      "data": [
        {
          "deep": 0.38333333333333336,
          "source": "Justin’s Apple Watch",
          "inBedStart": "2025-02-01 21:22:30 -0800",
          "asleep": 0,
          "date": "2025-02-02 05:30:30 -0800",
          "core": 4.8583333333333325,
          "rem": 1.6833333333333333,
          "inBedEnd": "2025-02-02 05:32:30 -0800",
          "sleepStart": "2025-02-01 21:22:30 -0800",
          "awake": 1.2416666666666667,
          "inBed": 0,
          "sleepEnd": "2025-02-02 05:32:30 -0800"
        }
      ],
      "name": "sleep_analysis",
      "units": "hr"
    }
   }`

**Comment by csmith** [comment]
For anyone that stumbles on this wanting to get the individual phase times exported, there's an answer from the author [on reddit](https://old.reddit.com/r/healthautoexport/comments/10730zu/export_sleep_stages/):

> I still need to update some guides around this but you can export sleep phases on the "Export" tab when you navigate to Browse All Data > Sleep > Sleep Analysis. There is then the option to turn "Aggregate sleep data" off. This should provide you with the various sleep phases. This option is also available when creating custom automations.
> 
> Note: This setting isn't available when exporting health metrics in bulk using the Quick Export action.

**Reply by MrRKernelPanic** [reply]
This is exactly what I cam here to find out about.  Is it possible to run an automation where the data is NOT aggregated?  The only options are default and hr.  This is the reason I wanted to get and use the app.  I have a cheap smartwatch that I can FORCE to push data to home assistant, but wanted an automated way of doing it.  The automations are working, just want the NOT aggregated sleep data for the different stages of sleep.  I have sleep Apnea so want to keep an eye on things.

**Reply by csmith** [reply]
I have an automation that does it. I selected _just_ the Sleep Analysis metric, and turned aggregation off:

![image](https://github.com/user-attachments/assets/0a2739c5-662e-426f-abf2-c28131c41bdf)

I don't remember why it needed to be a separate automation, but I do remember fiddling with it for a while before I got it working 😅

**Reply by MrRKernelPanic** [reply]
Interesting, I don't have the option to turn this off for automations, there seems no way to turn off Aggregation Interval, the only options are Default / Days on a drop down.  I can do this with a general EXPORT, but not for an automation.  I have tried doing this with just a the single 'select health metrics' (sleep analysis).  Can you confirm this export you have works for Home Assistant?  Have just noticed the option to turn this off seems to be available for REST-API, just not Home Assistant as the Automation Type.

**Reply by csmith** [reply]
Ah, I didn't realise the options changed when you changed automation type. Mine are all REST (I don't use HA). I see what you mean, I can't see a way to do it for HA :(
